### ISSUE TITLE
"odo delete --app" should throw error for invalid app in devfile

### ISSUE THREAD

**Issue by prietyc123** [openshift/odo#3493]
/kind bug

## What versions of software are you using?

**Operating System:**
Supported

**Output of `odo version`:**
master

## How did you run odo exactly?

```
$ odo create nodejs
Experimental mode is enabled, use at your own risk

Validation
 ✓  Checking devfile existence [45743ns]
 ✓  Checking devfile compatibility [107796ns]
 ✓  Creating a devfile component from registry: RegistryName [128026ns]
 ✓  Validating devfile component [452199ns]

Please use `odo push` command to create the component with source deployed

$ odo delete --app -h
? Are you sure you want to delete the devfile component: nodejs? No
 ✗  Aborting deletion of component

$ odo delete --app -h
? Are you sure you want to delete the devfile component: nodejs? Yes
 ✓  Deleting devfile component nodejs [904ms]
 ✓  Successfully deleted component

$ odo delete --all
? Are you sure you want to delete the devfile component: nodejs? Yes
 ✗  Deleting devfile component nodejs [314ms]
 ✗  error occurred while deleting component, cause: the component nodejs doesn't exist on the cluster
? Are you sure you want to delete env folder? No
 ✗  Aborting deletion of env folder

``` 

Tried with some more incorrect app name :

```
$ odo create nodejs
Experimental mode is enabled, use at your own risk

Validation
 ✓  Checking devfile existence [52347ns]
 ✓  Checking devfile compatibility [173289ns]
 ✓  Creating a devfile component from registry: RegistryName [177960ns]
 ✓  Validating devfile component [445600ns]

Please use `odo push` command to create the component with source deployed

$ odo delete --app incorrect-app
? Are you sure you want to delete the devfile component: nodejs? Yes
 ✗  Deleting devfile component nodejs [285ms]
 ✗  error occurred while deleting component, cause: the component nodejs doesn't exist on the cluster
```

## Actual behavior

`$ odo delete --app incorrect-app` asking to delete devfile component though the app is incorrect. 

## Expected behavior

`$ odo delete --app incorrect-app` it should directly throw error saying `Component nodejs does not exists on the cluster`

## Any logs, error output, etc?

Manually trying `odo delete --app` with experimental mode.

**Comment by mik-dass** [comment]
`App` is not yet implemented for devfiles https://github.com/openshift/odo/issues/3402

**Comment by prietyc123** [comment]
> App is not yet implemented for devfiles #3402

Great I was not aware of 😄
